peers/open_channels_for_saved_nodes.js

```diff
@@ -0,0 +1,830 @@
+              return cbk(err);
+            }
+              // Exit early when the peer is already connected
+              if (res.peers.map(n => n.public_key).includes(node.public_key)) {
+                return cbk();
+              }
+    
+              if (!node.sockets.length) {
+                return cbk([503, 'NoAddressFoundToConnectToNode', {node}]);
+              }
+
+            if (!!args.opening_nodes && !!args.opening_nodes.length) {
+                  args.logger.info({
+                    connecting_to: {alias: node.alias, public_key: node.public_key, from: node.opening_node_name},
+                  });
+              } else {
+                  args.logger.info({
+                  connecting_to: {alias: node.alias, public_key: node.public_key},
+                });
+              }
+
+              return asyncRetry({times}, cbk => {
+                return asyncDetectSeries(node.sockets, ({socket}, cbk) => {
+                  return addPeer({socket, lnd: node.opening_node_lnd, public_key: node.public_key}, err => {
+                    return cbk(null, !err);
+                  });
+                },
+                (err, res) => {
+                  if (!!err) {
+                    return cbk(err);
+                  }
+    
+                  if (!res) {
+                    return cbk([503, 'FailedToConnectToPeer', ({peer: node.public_key})]);
+                  }
+    
+                  return setTimeout(() => cbk(null, true), peerAddedDelayMs);
+                });
+              },
+              cbk);
+          });
+        },
+        cbk);
+      }],
+
+      // Get the wallet version and check if it is compatible
+      getWalletVersion: ['validate', ({}, cbk) => {
+        return getWalletVersion({lnd: args.lnd}, (err, res) => {
+          if (!!err) {
+            return cbk([400, 'BackingLndCannotBeUsedToOpenChannels', {err}]);
+          }
+
+          return cbk(null, {version: res.version});
+        });
+      }],
+
+      // Check all nodes that they will allow an inbound channel
+      checkAcceptance: [
+        'capacities',
+        'connect',
+        'getNodes',
+        ({capacities, connect, getNodes}, cbk) => 
+      {
+        const {channels} = channelsFromArguments({
+          capacities,
+          addresses: args.cooperative_close_addresses,
+          gives: args.gives,
+          nodes: args.public_keys,
+          types: args.types,
+        });
+
+        return asyncEachSeries(getNodes, (node, cbk) => {
+          const to = node.public_key;
+          const channel = {
+            capacity: node.capacity || defaultChannelCapacity,
+            cooperative_close_address: node.coop_close_address || undefined,
+            give: !!node.give ? Number(node.give) : undefined,
+            partner_public_key: node.public_key,
+            is_private: !!node.type && node.type === 'private'
+          }
+
+          return openChannels({
+            channels: [channel],
```

Comment:
all nodes except the last one to call fundPendingChannels must set `is_avoiding_broadcast: true`

Or... in the case of multiple nodes we could set that on all nodes and then broadcast the funding ourselves separately